ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/ModuleControlApplication.java

```diff
@@ -291,5 +292,43 @@ else if (action.equalsIgnoreCase("stop"))
         return new ResponseEntity("Context Listeners state changed successfully!", HttpStatus.OK);
     }
 
+    @RequestMapping(method = RequestMethod.PUT,
+        value = "/activator/{moduleName}/{action}")
```

Comment:
this is not a great way for creating REST for put call.
/activator   should accept DTO(modulename , action)

bit like this 
```
 @RequestMapping(method = RequestMethod.PUT,
        value = "/startupMode")
    @PreAuthorize("hasAnyAuthority('ALL','WebServiceAdmin')")
    public ResponseEntity changeFlowStartupMode(
        @RequestBody ChangeFlowStartupModeDto changeFlowStartupModeDto)
    {
        String user = UserUtil.getUser();

        String startupType = changeFlowStartupModeDto.getStartupType();
        String moduleName = changeFlowStartupModeDto.getModuleName();
        String flowName = changeFlowStartupModeDto.getFlowName();
        String startupComment = changeFlowStartupModeDto.getComment();

        if ("manual".equalsIgnoreCase(startupType)
            || "automatic".equalsIgnoreCase(startupType)
            || "disabled".equalsIgnoreCase(startupType))
        {
            //crude check to ensure comment is supplied when disabling
            if (startupType.equalsIgnoreCase("disabled") && (startupComment == null || ""
                .equals(startupComment.trim())))
            {
                return new ResponseEntity(new ErrorDto("Comment must be provided when disabling Flow startup"), HttpStatus.BAD_REQUEST);
            }
            moduleService.setStartupType(moduleName, flowName, StartupType.valueOf(startupType.toUpperCase()), startupComment, user);
            moduleMetadataDashboardRestService.publish(this.moduleService.getModule(moduleName));
            return new ResponseEntity(HttpStatus.OK);
        }
        else{
            return new ResponseEntity(new ErrorDto("Invalid startupType["+startupType+"]."), HttpStatus.BAD_REQUEST);

        }
    }

```